stdlib/importlib/metadata/__init__.pyi

```diff
@@ -68,9 +82,33 @@ class EntryPoint(_EntryPointBase):
 
     def __hash__(self) -> int: ...
     def __eq__(self, other: object) -> bool: ...
+    if sys.version_info >= (3, 11):
+        def __lt__(self, other: object) -> bool: ...
+    if sys.version_info < (3, 12):
+        def __iter__(self) -> Iterator[Any]: ...  # result of iter((str, Self)), really
```

Comment:
So why not use `Iterator[tuple[str, Self]]`?

Reply:
That's an iterator that returns `tuple[str, Self]`. This is an iterator that will always return two items, the first of which is str and the second of which is Self. As far as I know, there's no way to express that in the current system. We could do `Iterator[str | Self]`, but my understanding is that it's better to avoid union return types.

The implementation here is `return iter((self.name, self))`; I meant "iter()` as the runtime function, not a misspelling of the type, but that's a bit confusing because then I switched to the types of the values. Maybe the comment could be better worded, but I'm not sure how without getting too wordy. Might be better to just leave off the comment if it's too confusing. The function was removed in 3.12, so this is going away (eventually) anyway.